### PR TITLE
feat(worker/lido): support lido transactions

### LINKED ISSUE #68: lido
### Tag

Exchange

### Project URL

https://lido.fi/

### Worker description

Lido Protocol, Build web3 staking service.

### What actions should be included?

1. Staked ETH Submitted
2. Staked ETH Withdrawal NFT Withdrawal (Requested, Claimed)
3. Staked MATIC Submit
4. Staked MATIC Request Withdraw
5. Staked MATIC Claim Tokens
6. Staked ETH Transfer Shares

### Additional description

#71

### This is not a duplicated worker request

- [x] I have searched [existing issues](https://github.com/NaturalSelectionLabs/RSS3-Node/issues) and [pull requests](https://github.com/NaturalSelectionLabs/RSS3-Node/pulls) to ensure this worker proposal has not already been requested

### PR DESCRIPTION
**Pull request by Candinya**
resolve #68

### CODE REVIEW

File: deploy/dev/values.yaml [hunk 1]
```diff
@@ -80,6 +80,11 @@ indexers:
     parameters:
       block_height_start: 1275799
       rpc_thread_blocks: 1
+  - network: ethereum
+    worker: lido
+    endpoint: <path:kv/data/network/rss3-node#ENDPOINT_ETHEREUM>
+    parameters:
+      block_number_start: 17571053
```
Comment (kallydev) [inline deploy/dev/values.yaml:87]: Why is block number `17571053`?

File: deploy/prod/values.yaml [hunk 2]
```diff
@@ -64,6 +64,11 @@ indexers:
     parameters:
       block_height_start: 1275799
       rpc_thread_blocks: 1
+  - network: ethereum
+    worker: lido
+    endpoint: <path:kv/data/network/rss3-node#ENDPOINT_ETHEREUM>
+    parameters:
+      block_number_start: 17571053
```
Comment (kallydev) [inline deploy/prod/values.yaml:71]: > Why is block number 17571053?

And there.

File: deploy/dev/values.yaml (hunk 1 above)
Comment (Candinya) [reply deploy/dev/values.yaml]: fixed, using 11473216 for now ( `AddressStakedETH (0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84)` 's deploy block height)

File: internal/engine/worker/contract/lido/worker.go [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+		}
+
+		feed.Actions = append(feed.Actions, actions...)
+	}
+
+	return feed, nil
+}
+
+// NewWorker creates a new Lido worker.
+func NewWorker(config *config.Module) (engine.Worker, error) {
```
Comment (pseudoyu) [inline internal/engine/worker/contract/lido/worker.go:138]: 1. `NewWorker` should be the last function in `worker.go`
2. Should add more comments of match/transform function

**Review by pseudoyu — APPROVED** [review]

**Pull request merged**